src/main/java/org/elasticsearch/index/query/BaseQueryBuilder.java

```diff
@@ -68,5 +69,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public abstract Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException;
```

Comment:
sorry, javadocs should go the interface, I think you could remove this method from the base abstract class.